src/mpid/ch3/channels/nemesis/netmod/tcp/tcp_init.c

```diff
@@ -126,6 +126,10 @@ static int set_up_listener(void)
 
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_SET_UP_LISTENER);
 
+	/*IPV6
+	if (ipv6_flag_set)
+	PID_nem_tcp_g_lstn_plfd.fd = MPID_nem_tcp_g_lstn_sc.fd = socket(AF_INE T6, SOCK_STREAM, IPPROTO_TCP);
+	else */
```

Comment:
Remove this.

Reply:
Should try to open to an IPv4 socket first, only open to IPv6 socket when IPv4 is not available. This keeps the existing behavior if a dual-stack network is present. Since this is almost the first function to be called during initialization, you can set the global flag here.